src/main/java/CliUI.java

```diff
@@ -1,87 +1,153 @@
+            } break;
+            default: break;
+        }
     }
 
     void uiLoop() {
         while (isOpen) {
             displayMenu();
-            getAndExecuteCommand();
+            executeCommand();
         }
     }
 
     private void displayMenu() {
-        System.out.println("Menu:\n1. View all snippets\n2. Create a snippet\n3. Delete a snippet\n4. Edit a snippet\n5. Filter\n6. Quit");
+        System.out.println(
+                "Menu:\n" +
+                "1. View all snippets\n" +
+                "2. Create a snippet\n" +
+                "3. Delete a snippet\n" +
+                "4. Edit a snippet\n" +
+                "5. Filter\n" +
+                "6. Search\n" +
+                "7. Quit");
     }
 
-    private void getAndExecuteCommand() {
-        int input = keyboard.nextInt();
+    private void executeCommand() {
+        int input = getIntInput();
 
         switch (input) {
             case 1: System.out.println(snippetManager.listAll()); break;
             case 2: createSnippet(); break;
             case 3: deleteSnippet(); break;
             case 4: editSnippet(); break;
             case 5: filterSnippets(); break;
-            case 6: isOpen = false; break;
+            case 6: searchSnippets(); break;
+            case 7: isOpen = false; break;
             default: System.out.println(input + ": please select a valid option");
         }
     }
 
     private void createSnippet() {
         System.out.println("Enter the title of the snippet:");
-        String title = keyboard.next();
+        String title = keyboard.nextLine();
 
         System.out.println("Enter the language of the snippet:");
-        String lang = keyboard.next();
+        String lang = keyboard.nextLine();
 
         System.out.println("Enter the tags of the snippet, like tag1,tag2,...:");
-        String[] tags = keyboard.next().split(",");
+        String[] tags = keyboard.nextLine().split(",");
 
-        Integer snippetId = snippetManager.create(title, "", lang, tags);
-        snippetManager.edit(snippetId);
+        snippetManager.create(title, null, lang, tags);
     }
 
     private void deleteSnippet() {
         System.out.println("Enter the id of the snippet to delete:");
-        int input = keyboard.nextInt();
+        int input = getIntInput();
         snippetManager.delete(input);
     }
 
     private void editSnippet() {
         System.out.println("Enter the id of the snippet to edit");
-        Integer snippetId = keyboard.nextInt();
+        Integer snippetId = getIntInput();
         snippetManager.edit(snippetId);
     }
 
     private void filterSnippets() {
-        System.out.println("Enter the title to filter by (... for none):");
-        String title = keyboard.next();
-        if (title.equals("...")) {
-            title = "";
+        System.out.println("Enter the term to filter by (... for none):");
+        String filterTerm = keyboard.nextLine();
+        if (filterTerm.equals("...")) {
```

Comment:
Dots are kind of tedious to type... in my opinion. Maybe just check for an empty line instead of three dots.